timesketch/lib/analyzers/safebrowsing.py

```diff
@@ -0,0 +1,250 @@
+                        if key not in self._SAFEBROWSING_ENTRY_KEEP:
+                            threat_result.pop(key)
+
+                    results[match['threat']['url']] = threat_result
+
+        return results
+
+    def _sanitize_url(self, url_entry):
+        """Finds http[s]:// in 'url_entry' and returns its content from there.
+
+        Args:
+            url_entry: a URL, with some other characters before and after
+
+        Returns:
+            String with the URL
+        """
+        m = self._URL_BEGINNING_RE.search(url_entry)
+
+        if m:
+            return m.group(1)
+
+        return None
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query = (
+            '{"query": { "bool": { "should": [ '
+            '{ "exists" : { "field" : "url" }} ] } } }')
+
+        return_fields = ['url']
+
+        events = self.event_stream(
+            query_dsl=query,
+            return_fields=return_fields,
+        )
+
+        urls = {}
+
+        for event in events:
+            url = self._sanitize_url(event.source.get('url'))
+
+            if not url:
+                continue
+
+            urls.setdefault(url, []).append(event)
+
+        # Exit early if there are no URLs in the data set to analyze.
+        if not urls:
+            return 'No URLs to analyze.'
+
+        if not self._safebrowsing_api_key:
+            return 'Safe Browsing API requires an API key!'
+
+        url_whitelisted = 0
+
+        url_whitelist = set(
+            interface.get_yaml_config(
+                self._URL_WHITELIST_CONFIG,
+            ),
+        )
+
+        if not url_whitelist:
+            domain_analyzer_whitelisted = current_app.config.get(
+                'DOMAIN_ANALYZER_WHITELISTED_DOMAINS',
+                [],
+            )
+            for domain in domain_analyzer_whitelisted:
+                url_whitelist.add('*.%s/*' % domain)
+
+        logging.info(
+            '{:d} entries on the whitelist.'.format(len(url_whitelist)),
+        )
+
+        safebrowsing_platforms = current_app.config.get(
+            'SAFEBROWSING_PLATFORMS',
+            ['ANY_PLATFORM'],
+        )
+
+        safebrowsing_types = current_app.config.get(
+            'SAFEBROWSING_THREATTYPES',
+            ['MALWARE'],
+        )
+
+        lookup_urls = []
+
+        for url in urls:
+            if self._is_url_whitelisted(url, url_whitelist):
+                url_whitelisted += 1
+                continue
+
+            lookup_urls.append(url)
+
+        try:
+            safebrowsing_results = self._do_safebrowsing_lookup(
+                lookup_urls,
+                safebrowsing_platforms,
+                safebrowsing_types,
+            )
+        except requests.HTTPError:
+            return "Couldn't reach the Safe Browsing API."
+
+        for url, events in urls.items():
```

Comment:
shouldn't this be `for url.... in lookup_urls` since those are the only ones you actually did a lookup on.. no need to check the whitelisted ones here, is there?

Reply:
a gentle ping about this one

Reply:
No, both are correct. Changing it anyway!